bridge-history-api/internal/orm/cross_message.go

```diff
@@ -217,6 +217,14 @@ func (c *CrossMessage) UpdateL1MessageQueueEventsInfo(ctx context.Context, l1Mes
 			db = db.Where("message_nonce = ?", l1MessageQueueEvent.QueueIndex)
 			db = db.Where("message_type = ?", btypes.MessageTypeL1SentMessage)
 			txStatusUpdateFields["tx_status"] = types.TxStatusTypeDropped
+		case btypes.MessageQueueEventTypeResetDequeuedTransaction:
+			// do not over-write terminal statuses.
+			db = db.Where("tx_status != ?", types.TxStatusTypeRelayed)
+			db = db.Where("tx_status != ?", types.TxStatusTypeDropped)
+			// reset skipped messages that the nonce is greater than or equal to the queue index.
+			db = db.Where("message_nonce >= ?", l1MessageQueueEvent.QueueIndex)
+			db = db.Where("message_type = ?", btypes.MessageTypeL1SentMessage)
+			txStatusUpdateFields["tx_status"] = types.TxStatusTypeSent
```

Comment:
it's better test this sql if it can use the index

Reply:
no, it doesn't. example:
```
explain analyze select * from cross_message_v2 where tx_status != 2 and tx_status != 6 and message_nonce >= 100000 and message_type != 1;

Gather  (cost=1000.00..89184.20 rows=4986 width=947) (actual time=12.637..101.645 rows=6537 loops=1)
  Workers Planned: 2
  Workers Launched: 2
  ->  Parallel Seq Scan on cross_message_v2  (cost=0.00..87685.60 rows=2078 width=947) (actual time=7.429..97.113 rows=2179 loops=3)
        Filter: ((tx_status <> 2) AND (tx_status <> 6) AND (message_nonce >= 100000) AND (message_type <> 1))
        Rows Removed by Filter: 359146
Planning Time: 0.128 ms
Execution Time: 101.981 ms
```

When handling `btypes.MessageQueueEventTypeDequeueTransaction` and `btypes.MessageQueueEventTypeResetDequeuedTransaction`, both do not have any index to use. but these events are very unlikely to happen, do we still add an index to optimize them?

Reply:
after changing the sql in [46d50a4](https://github.com/scroll-tech/scroll/pull/1467/commits/46d50a4153b45c65249495cf8cd24ad9f82d94ac), it can use index:
```
EXPLAIN ANALYZE
SELECT *
FROM cross_message_v2
WHERE tx_status = 5 AND message_nonce >= 777291 AND message_type = 1;
```